airflow/dag_processing/processor.py

```diff
@@ -375,16 +375,15 @@ def manage_slas(self, dag: DAG, session: Session = None) -> None:
             .group_by(TI.task_id)
             .subquery('sq')
         )
-        # get recorded SlaMiss
-        recorded_slas_query = set(
-            session.query(SlaMiss.dag_id, SlaMiss.task_id, SlaMiss.execution_date).filter(
+        recorded_sla_misses = set(
+            session.query(SlaMiss.dag_id, SlaMiss.task_id, SlaMiss.run_id, SlaMiss.map_index).filter(
                 SlaMiss.dag_id == dag.dag_id, SlaMiss.task_id.in_(dag.task_ids)
             )
         )
 
         max_tis: Iterator[TI] = (
             session.query(TI)
-            .join(TI.dag_run)
+            .join(TI.dag_run)  # todo: why?
```

Comment:
I believe this can be removed now. The `lazy="joined"` setting on `TI.dag_run` is a recent-ish addition, and this is likely added before that to prevent an N+1 query issue in the `for ti in max_tis` loop below that accesses `ti.dag_run`.

Reply:
you think i should leave it for another PR or just lump it in?

Reply:
No, just delete the line in this PR.